pennylane/plugins/default_gaussian.py

```diff
@@ -371,6 +372,24 @@ def controlled_phase(s):
     return S
 
 
+def interferometer(U):
+    """Interferometer
+
+    Args:
+        U (array): unitary matrix.
+
+    Returns:
+        array: symplectic transformation matrix.
+    """
+    N = 2*len(U)
+    X = U.real
+    Y = U.imag
+    rows = np.arange(N).reshape(2, -1).T.flatten()
+    S = np.vstack([np.hstack([X, -Y]),
+                   np.hstack([Y, X])])[:, rows][rows]
+
+    return S
+
```

Comment:
An alternative to implementing this here again, would be to leverage `ops.Interferometer._heisenberg_rep()` to construct S. The upside would be reduced code duplication, the downside a slightly odd dependence of the plugin on a "private" method of a PL Operation. At first sight, for me the advantages are greater than the disadvantages: If ever incompatible changes are made to `Interferometer`, then it will probably result in an error of some test. However, if a bug is found and fixed in  `ops.Interferometer._heisenberg_rep()` then the same bug here might remain unfixed...

Reply:
My preference is to have the plugins as independent of private PL methods as possible. Primarily because the default plugins are used as 'reference plugins', and developers are directed to them multiple times in the docs as a reference.

Even the way the plugin submodule is accessed is separate to the rest of PennyLane core - it is never accessed directly as part of the same package, instead they can only be accessed after running `setup.py` and creating the correct entry points.

As a result, even though it is usually preferable to avoid code duplication, in this case there is a bit more of a separation between `pennylane.plugins` and the remainder of PennyLane core. Coupled with the fact that they exist for developer reference, I think it is better to be explicit here and not depend on core private methods.

Reply:
Ok, I understand that.